checkov/terraform/checks/resource/gcp/CloudArmorWAFACLCVE202144228.py

```diff
@@ -24,14 +24,17 @@ def scan_resource_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
             ]
             match = rule.get("match")
             if match:
-                expr = match[0].get("expr")
-                if expr and expr[0].get("expression") == ["evaluatePreconfiguredExpr('cve-canary')"]:
-                    if rule.get("action") == ["allow"]:
-                        return CheckResult.FAILED
-                    if rule.get("preview") == [True]:
-                        return CheckResult.FAILED
-
-                    return CheckResult.PASSED
+                try:
+                    expr = match[0].get("expr")
+                    if expr and expr[0].get("expression") == ["evaluatePreconfiguredExpr('cve-canary')"]:
+                        if rule.get("action") == ["allow"]:
+                            return CheckResult.FAILED
+                        if rule.get("preview") == [True]:
+                            return CheckResult.FAILED
+
+                        return CheckResult.PASSED
+                except Exception:
+                    return CheckResult.UNKNOWN
```

Comment:
can you check the specific edge case? and add test for it :)

Reply:
that would be great, I'm also curious what exactly is failing here and why.

Reply:
the case was that `rule` is a list, so the `try-except` catches this case of unexpected type and also unexpected `expr` type, in case there will be